app/src/main/scala/org/alephium/explorer/persistence/model/CreateSubContractEventEntity.scala

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see <http://www.gnu.org/licenses/>.
+
+package org.alephium.explorer.persistence.model
+
+import org.alephium.api.model.ValAddress
+import org.alephium.protocol
+import org.alephium.protocol.model.{Address, BlockHash, TransactionId}
+import org.alephium.util.TimeStamp
+
+final case class CreateSubContractEventEntity(
+    blockHash: BlockHash,
+    txHash: TransactionId,
+    contract: Address,
+    subContract: Address,
+    timestamp: TimeStamp,
+    eventOrder: Int
+)
```

Comment:
Should we change this as:
```
  parentContract: Option[Address],
  newContract: Address,
```

In this way, we could store all of the contract creation events, not just for subcontracts.
Later, we could use the same database to query how many contracts created in total.

Reply:
this is already stored in the [global event table](https://github.com/alephium/explorer-backend/blob/feature/contract-subcontract/app/src/main/scala/org/alephium/explorer/persistence/schema/EventSchema.scala). The idea here was to have a specific table for the sub-contract creation for fast queries.

Reply:
oh but your idea was to have a table only for `Contract Creation` events?

Reply:
Probably a table for `Contract` makes more sense. The fields could be:
```
Contract(
  address: Address, parentContract: Address,
  creationTS: TimeStamp, creationBlock: BlockHash, ..., // all creation columns
  destructionTs: TimeStamp, destructionBlock: BlockHash, ... // all destruction columns.
)
```

Reply:
In the future, we may insert more data for each contract